### PR TITLE
controller: updated documentation for version check

### PR DESCRIPTION
**Pull request by Tbaile**
Updating documentation regarding on how `Version Checking` works and how to bypass it

### CODE REVIEW

File: controller.rst [hunk 1]
```diff
@@ -262,4 +264,95 @@ Controller without subscription:
 Controller with a valid subscription:
 
 - The number of units is unlimited.
-- Only firewalls with a valid subscription can register with the controller.
\ No newline at end of file
+- Only firewalls with a valid subscription can register with the controller.
+
+Version Awareness
```
Comment (gsanchietti) [inline controller.rst:269]:
```suggestion
Version awareness
```

File: controller.rst [hunk 2]
```diff
@@ -262,4 +264,95 @@ Controller without subscription:
...
+Version awareness is a mechanism that prevents the user from performing operations that are not supported by the unit version. To do so, when connecting to the UI
+of a unit the controller will ask for the API version during the connection process. There are three possible scenarios:
+
+a. If the versions are compatible, the connection proceeds as normal.
+b. If the firewall (unit) is significantly older than the controller, you'll see a popup that prevents the connection. This is to protect against potential errors.
+c. If the controller is slightly older than the firewall, you'll see a warning about the mismatch. However, you'll still be able to connect if you choose to proceed.
+
+As an administrator, you don't need to take any specific actions to enable Version awareness. It works automatically in the background. However, you should:
+
+1. Pay attention to warnings: If you see a version mismatch warning, consider updating your system when convenient.
```
Comment (gsanchietti) [inline controller.rst:281]:
```suggestion
1. Pay attention to warnings: if you see a version mismatch warning, consider updating your system when convenient.
```

File: controller.rst [hunk 3]
```diff
@@ -262,4 +264,95 @@ Controller without subscription:
...
+2. Keep your system updated: Regularly check for and apply updates to both your controller and firewall units to ensure the best compatibility and access to new features.
```
Comment (gsanchietti) [inline controller.rst:282]:
```suggestion
2. Keep your system updated: regularly check for and apply updates to both your controller and firewall units to ensure the best compatibility and access to new features.
```

File: controller.rst [hunk 4]
```diff
@@ -262,4 +264,95 @@ Controller without subscription:
...
+3. Report issues: If you encounter any unusual behavior or errors, especially after seeing a version warning, report it to your IT support team or Nethsecurity support.
```
Comment (gsanchietti) [inline controller.rst:283]: I'd add link to https://docs.nethsecurity.org/en/latest/troubleshooting.html:
```suggestion
3. Report issues: if you encounter any unusual behavior or errors, especially after seeing a version warning, follow the troubleshooting procedure
```

File: controller.rst [hunk 5]
```diff
@@ -262,4 +264,95 @@ Controller without subscription:
...
+
+Version awareness is a behind-the-scenes feature that helps keep your Nethsecurity system running smoothly. By automatically checking compatibility between the controller
```
Comment (gsanchietti) [inline controller.rst:285]:
```suggestion
Version awareness is a behind-the-scenes feature that helps keep your NethSecurity system running smoothly. By automatically checking compatibility between the controller
```

File: controller.rst [hunk 6]
```diff
@@ -262,4 +264,95 @@ Controller without subscription:
...
 - The number of units is unlimited.
-- Only firewalls with a valid subscription can register with the controller.
\ No newline at end of file
+- Only firewalls with a valid subscription can register with the controller.
+
+Version Awareness
+=================
+
+Version awareness is a mechanism that prevents the user from performing operations that are not supported by the unit version. To do so, when connecting to the UI
+of a unit the controller will ask for the API version during the connection process. There are three possible scenarios:
```
Comment (gsanchietti) [inline controller.rst:273]:
```suggestion
of a unit the controller will check the API version during the connection process. There are three possible scenarios:
```

File: controller.rst [hunk 7]
```diff
@@ -262,4 +264,95 @@ Controller without subscription:
...
+
+b. To update the image, you can simply schedule the installation, remember this is a operation that restarts the firewall (causing a downtime)
+
+   1. Check if there is an updated image available:
+ 
+      .. code-block:: bash
+ 
+        /usr/libexec/rpcd/ns.update call check-system-update
+
+   2. If you want to proceed with the update, you must first generate a timestamp for the update, this can be done with the following command:
```
Comment (gsanchietti) [inline controller.rst:331]: Scheduling from command line is a bit tricky.
Why not documenting only the immediate upgrade procuder?

Comment (Tbaile) [reply controller.rst:331]: I figured this would come up. And I agree :+1:

File: controller.rst [hunk 8]
```diff
@@ -262,4 +264,72 @@ Controller without subscription:
...
+As an administrator, you don't need to take any specific actions to enable Version awareness. It works automatically in the background. However, you should:
+
+1. Pay attention to warnings: if you see a version mismatch warning, consider updating your system when convenient.
+2. Keep your system updated: regularly check for and apply updates to both your controller and firewall units to ensure the best compatibility and access to new features.
+3. Report issues: if you encounter any unusual behavior or errors, especially after seeing a version warning, follow the troubleshooting procedure
+
+Version awareness is a behind-the-scenes feature that helps keep your NethSecurity system running smoothly. By automatically checking compatibility between the controller
+and units, it prevents many potential issues before they can affect your network. While it doesn't require any action from you, being aware of this feature can help you better understand and manage your system.
+
+Remember, if you're ever unsure about a version mismatch or update, it's always best to consult with your IT department or Nethsecurity support for guidance.
```
Comment (gsanchietti) [inline controller.rst:288]:
```suggestion
```

File: controller.rst [hunk 9]
```diff
@@ -262,4 +264,72 @@ Controller without subscription:
...
+
+a. If the versions are compatible, the connection proceeds as normal.
+b. If the firewall (unit) is significantly older than the controller, you'll see a popup that prevents the connection. This is to protect against potential errors.
+c. If the controller is slightly older than the firewall, you'll see a warning about the mismatch. However, you'll still be able to connect if you choose to proceed.
+
+As an administrator, you don't need to take any specific actions to enable Version awareness. It works automatically in the background. However, you should:
+
+1. Pay attention to warnings: if you see a version mismatch warning, consider updating your system when convenient.
+2. Keep your system updated: regularly check for and apply updates to both your controller and firewall units to ensure the best compatibility and access to new features.
+3. Report issues: if you encounter any unusual behavior or errors, especially after seeing a version warning, follow the troubleshooting procedure
```
Comment (gsanchietti) [inline controller.rst:283]: Add the link to troubleshooting section

**Review by gsanchietti — APPROVED** [review]

**Pull request merged**